massa-api/src/public.rs

```diff
@@ -381,11 +381,46 @@ impl MassaRpcServer for API<Public> {
             })
             .collect::<BTreeMap<_, _>>();
 
+        let current_cycle = last_slot
+            .unwrap_or_else(|| Slot::new(0, 0))
+            .get_cycle(api_settings.periods_per_cycle);
+
+        let cycle_duration = match api_settings.t0.checked_mul(api_settings.periods_per_cycle) {
+            Ok(cycle_duration) => cycle_duration,
+            Err(e) => return Err(ApiError::TimeError(e).into()),
+        };
+
+        let previous_cycle = current_cycle.saturating_sub(1);
+
+        let elapsed_time_before_current_cycle = match if previous_cycle == 0 {
```

Comment:
I think you should destruct this pattern. I find the `match if` pretty hard to read